airflow/jobs/scheduler_job_runner.py

```diff
@@ -774,9 +774,11 @@ def _process_executor_events(self, session: Session) -> int:
                     tags={"dag_id": ti.dag_id, "task_id": ti.task_id},
                 )
                 msg = (
-                    "Executor reports task instance %s finished (%s) although the "
-                    "task says it's %s. (Info: %s) Was the task killed externally?"
+                    f"The executor reported that the task instance {ti} finished with state {state}, but the task instance's state attribute is {ti.state}. "
+                    "Learn more: https://airflow.apache.org/docs/apache-airflow/stable/troubleshooting.html#task-state-changed-externally"
                 )
```

Comment:
Don’t use f-string in logs; use the % format syntax as previously instead.

Reply:
IIRC a pre-commit hook updated this one. I naively assumed that meant Python had optimized f-strings to be more efficient in logging 😅